### DISCUSSION TITLE
How To Get Current FCM Token?

### DISCUSSION THREAD

**Discussion by MichaelRodrigueess0s1** [shinyorg/shiny#895] [Q&A] [answered]
Hello guys, has a way to get  to get current FMC deviceId using shiny.firebase? because i need send a push notification to specifics devices.

**Comment by aritchie** [comment, marked as answer]
Please try to look at the source code before asking questions like this.  You don't have to dig far at all to get this answer.

```csharp
var pushManager = ShinyHost.Resolve<IPushManager>();
var result = await pushManager.RequestAccess();
if (result.Status == AccessStatus.Available) {
    result.RegistrationToken //<= firebase token
}

// OR

pushManager.CurrentRegistrationToken 
```

**Reply by MichaelRodrigueess0s1** [reply]
thank you so much my dear!! your project is really amazing, i hope i can finish this app i'm developing to become a project sponsor.
